src/Core/Silk.NET.Core/Miscellaneous/Bool32.cs

```diff
@@ -1,18 +1,56 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
+
 namespace Silk.NET.Core
 {
     /// <summary>
     /// A 32-bit boolean.
     /// </summary>
-    public readonly struct Bool32
+    public readonly struct Bool32 : IEquatable<Bool32>, IEquatable<uint>, IEquatable<bool>
     {
         /// <summary>
         /// Gets the 32-bit value for this boolean.
         /// </summary>
         public uint Value { get; }
 
+        /// <summary>
+        /// Compares the current <see cref="Bool32"/> to another <see cref="object"/>. Returns true if they are equal.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns>True if <paramref name="obj"/> is equal to the current <see cref="Bool32"/>. 
+        /// This will always be false if <paramref name="obj"/> is not a <see cref="Bool32"/>.</returns>
+        public override bool Equals(object obj)
+        {
+            if (obj is Bool32 other)
+                return Equals(other);
```

Comment:
Should this also check for the other types so this just acts as a generic trampoline? i.e.:
```cs
public override bool Equals(object obj) => obj switch
{
    Bool32 val => Equals(val),
    bool val => Equals(val),
    uint val => Equals(val),
    _ => base.Equals(obj)
}
```

Reply:
This seems like a good idea. Makes it consistent with the explicit Equals() overloads.